src/main/java/com/exadel/sandbox/role/entity/Role.java

```diff
@@ -0,0 +1,46 @@
+package com.exadel.sandbox.role.entity;
+
+import com.exadel.sandbox.permission.entity.Permission;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Entity
+@EntityListeners(AuditingEntityListener.class)
+public class Role {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "role_id")
+    private Integer id;
+
+
+    @Column(name = "role_name", nullable = false, unique = true)
+    private String name;
+
+
+    @CreatedDate
+    @Column(name = "role_created")
+    private Date created;
+
+
+    @LastModifiedDate
+    @Column(name = "role_modified")
+    private Date modified;
+
+    @ManyToMany(cascade = {CascadeType.ALL})
```

Comment:
Should it be manyToMany? I think one role can have many permissions

Reply:
Do you prefer to have oneToMany relation here.? 
I just thought there can be a single permission in multiple roles, just as there can be multiple permissions in a single role.